core/src/smtp/mod.rs

```diff
@@ -366,14 +366,38 @@ async fn create_smtp_future(
 			is_disabled: false,
 		}
 	} else {
-		email_deliverable(&mut smtp_client, to_email).await?
+		let mut result = email_deliverable(&mut smtp_client, to_email).await;
+
+		// Some SMTP servers automatically close the connection after an error,
+		// so we should reconnect to perform a next command.
+		//
+		// Unfortunately `smtp_client.is_connected()` doesn't report about this,
+		// so we can only check for "io: incomplete" SMTP error being returned.
+		// https://github.com/async-email/async-smtp/issues/37
+		if is_io_incomplete_smtp_error(&result) {
```

Comment:
Shouldn't we do this check also in `smtp_is_catch_all`?

Reply:
@amaurymartiny I don't think so. This can happen in `smtp_is_catch_all` only if previous `MAIL FROM` command fails. But if that happens we have an early return and don't attempt `smtp_is_catch_all`.

Reply:
You're right 👍